contracts/ics/packet/src/entry.rs

```diff
@@ -1,229 +1,13 @@
-}
-
-#[inline]
-fn load_envelope() -> Result<Envelope> {
-    let witness_len = hl::load_transaction()?.witnesses().len();
-    let last_witness = hl::load_witness_args(witness_len - 1, Source::Input)?;
-    let envelope_data = last_witness.output_type();
-    if envelope_data.is_none() {
-        return Err(Error::WitnessIsIncorrect);
-    }
-    let envelope_bytes = envelope_data.to_opt().unwrap();
-    let envelope_slice = &envelope_bytes.raw_data();
-    decode::<Envelope>(envelope_slice).map_err(|_| Error::EnvelopeEncoding)
-}
-
-#[inline]
-fn load_and_validate_ibc_packet_from_idx(
-    idx: usize,
-    source: Source,
-) -> Result<(IbcPacket, PacketArgs)> {
-    let witness = hl::load_witness_args(idx, source)?;
-
-    let cell_data = if source == Source::Input {
-        witness.input_type()
-    } else {
-        witness.output_type()
+use ics_base::error::{CkbResult, Error};
+use ics_base::handler::{navigate_packet, verify, Navigator};
+use ics_base::utils::load_client;
+
+pub fn main() -> CkbResult<()> {
+    let envelope = match navigate_packet()? {
+        Navigator::CheckMessage(envelope) => envelope,
+        _ => return Err(Error::UnexpectedPacketMsg.into()),
     };
 
-    if cell_data.is_none() {
-        return Err(Error::WitnessIsNotExisted);
-    }
-
-    let bytes = cell_data.to_opt().unwrap();
-    let slice = bytes.as_slice();
-
-    let data = hl::load_cell_data(1, source)?;
-    let expected_hash: [u8; 32] = data.try_into().map_err(|_| Error::ChannelHashUnmatch)?;
-
-    if keccak256(slice) != expected_hash {
-        return Err(Error::ChannelHashUnmatch);
-    }
-
-    let lock = hl::load_cell_lock(idx, source)?;
-    let lock_args = lock.args().raw_data();
-    let packet_args = PacketArgs::from_slice(&lock_args).map_err(|_| Error::PacketLock)?;
-    let packet = decode_ibc_packet(slice)?;
-    Ok((packet, packet_args))
-}
-
-#[inline]
-fn decode_ibc_packet(bytes: &[u8]) -> Result<IbcPacket> {
-    decode(bytes).map_err(|_| Error::PacketEncoding)
-}
-
-#[inline]
-fn decode_channel_cell(bytes: &[u8]) -> Result<IbcChannel> {
-    decode(bytes).map_err(|_| Error::ChannelEncoding)
-}
-
-fn keccak256(slice: &[u8]) -> [u8; 32] {
-    let mut hasher = Keccak::v256();
-    hasher.update(slice);
-    let mut output = [0u8; 32];
-    hasher.finalize(&mut output);
-    output
-}
-
-fn load_client() -> Result<Client> {
-    let metadata =
-        hl::load_cell_data(0, Source::CellDep).map_err(|_| Error::FailedToLoadClientCellData)?;
-    let metadata_type_script = hl::load_cell_type(0, Source::CellDep)
-        .map_err(|_| Error::FailedToLoadClientTypeScript)?
-        .unwrap();
-    let client_data = metadata_type_script.args().raw_data();
-    let client_id: [u8; 32] = client_data
-        .as_ref()
-        .try_into()
-        .map_err(|_| Error::FailedToLoadClientId)?;
-    Client::new(client_id, &metadata).map_err(|_| Error::FailedToCreateClient)
+    let client = load_client().unwrap_or_default();
```

Comment:
Not sure if we can use `unwrap_or_default` here.
It is generally recommended to avoid using `unwrap_or_default`.

Reply:
it can be eliminated now cause I adjusted the error handling process after I wrote it